include/deal.II/lac/precondition.h

```diff
@@ -2451,44 +2451,52 @@ namespace internal
         internal::set_initial_guess(temp_vector1);
         data.constraints.set_zero(temp_vector1);
 
-        if (data.eigenvalue_algorithm == internal::EigenvalueAlgorithm::lanczos)
+        // note: temp_vector1 only contains zeroes, which happens if all dofs
+        // are constrained, there is nothing to do
+        if (temp_vector1.l2_norm() != 0.0)
```

Comment:
I think the comment has the wrong grammar. "`temp_vector1` only contains zeroes" is a statement, but the statement is not generally true. I think what you want to say is something like this:
```suggestion
        // We want to run the power iteration starting with temp_vector1, but for that it
        // needs to be a nonzero vector. However, there are situations where it only contains
        // zeroes (e.g., if all dofs are constrained). If that is the case, there is nothing to do.
        if (temp_vector1.l2_norm() != 0.0)
```